agent/internal/task_config.go

```diff
@@ -19,14 +19,25 @@ import (
 )
 
 type TaskConfig struct {
-	Distro             *apimodels.DistroView
-	ProjectRef         model.ProjectRef
-	Project            model.Project
-	Task               task.Task
-	BuildVariant       model.BuildVariant
-	Expansions         util.Expansions
-	NewExpansions      *agentutil.DynamicExpansions
-	DynamicExpansions  util.Expansions
+	Distro       *apimodels.DistroView
+	ProjectRef   model.ProjectRef
+	Project      model.Project
+	Task         task.Task
+	BuildVariant model.BuildVariant
+
+	// Expansions store the fundamental expansions set by Evergreen.
+	// e.g. execution, project_id, task_id, etc. It also stores
+	// expansions that are set by the user by expansion.update.
+	Expansions util.Expansions
```

Comment:
Do we need this `Expansions` field anymore? It seems kind of superfluous, and I think might just be leftover tech debt from when the infra team implementing the redaction logic. Can we file a ticket to delete it?

I don't think we can do away with the `DynamicExpansions` field because those address a very specific issue where function vars ended up persisting outside of functions, and we wanted those function vars to only persist beyond the function if the function var was also set within the function by `expansions.update`. That being said, I feel like we should rename the `DynamicExpansions` struct because it's just confusing - we could call it something else, like `SafeExpansions` to make it clearer that it's a thread-safe implementation. Feel free to open a separate ticket to untangle the expansion fields/types if you think it'll help.

Reply:
I was thinking this same thing, I'll create a ticket to remove it/rename dynamic so it's not bundled in this one!